src/main/kotlin/com/github/psxpaul/task/AbstractExecFork.kt

```diff
@@ -101,7 +100,7 @@ abstract class AbstractExecFork(objectFactory: ObjectFactory) : DefaultTask(), P
             }
             logger.info("Adding {} as a finalizing task to {}", joinTask.name, stopAfterValue.name)
             stopAfterValue.configure {
-                finalizedBy(joinTask)
+                it.finalizedBy(joinTask)
```

Comment:
This was the cause of most test failures as it referenced to the start task, which caused the task to be termined directly after it was started